core/src/bls_sigverify/stats.rs

```diff
@@ -0,0 +1,311 @@
+use std::{
+    sync::atomic::{AtomicU64, Ordering},
+    time::{Duration, Instant},
+};
+
+pub(crate) const STATS_INTERVAL_DURATION: Duration = Duration::from_secs(1);
+
+#[derive(Debug, Default)]
+pub(crate) struct BLSPacketStats {
+    pub(crate) recv_batches_us_hist: histogram::Histogram, // time to call recv_batch
+    pub(crate) verify_batches_pp_us_hist: histogram::Histogram, // per-packet time to call verify_batch
+    pub(crate) dedup_packets_pp_us_hist: histogram::Histogram, // per-packet time to call verify_batch
+    pub(crate) batches_hist: histogram::Histogram, // number of packet batches per verify call
+    pub(crate) packets_hist: histogram::Histogram, // number of packets per verify call
+    pub(crate) num_deduper_saturations: usize,
+    pub(crate) total_batches: usize,
+    pub(crate) total_packets: usize,
+    pub(crate) total_dedup: usize,
+    pub(crate) total_dedup_time_us: usize,
+    pub(crate) total_verify_time_us: usize,
```

Comment:
would be nice to be uniform across these stats struct and either add a `last_stats_logged` field here or evict it from `BLSSigVerifierStats`

Reply:
I mean that `BLSSigVerifierStats` has a `last_stats_logged` field inside the struct that holds the last time the stats were reported. This is then checked inside `fn maybe_report_stats` to see if we should report.

In contrast, `BLSPacketStats` has no such field. It instead relies on local variable `last_print` inside `fn verifier_service` to decide whether to call `fn maybe_report`.

We're effectively achieving the same thing with 2 different styles in the same crate. Consistency within the crate would be nice. My preference is to add a `last_stats_logged` field inside `BLSPacketStats` and then decide inside `fn maybe_report` if enough time has passed to report (and handle the reset and such). But I won't be picky as long as we pick one style to use across both